src/openrct2/localisation/Formatting.cpp

```diff
@@ -0,0 +1,207 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2020 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "Formatting.h"
+
+#include "../config/Config.h"
+#include "../util/Util.h"
+#include "FormatCodes.h"
+#include "Language.h"
+#include "StringIds.h"
+
+#include <cstdint>
+
+namespace OpenRCT2
+{
+    char GetDigitSeperator()
+    {
+        return ',';
+    }
+
+    char GetDecimalSeperator()
+    {
+        return '.';
+    }
+
+    template<size_t TDecimalPlace, bool TDigitSep, typename T> void FormatNumber(std::stringstream& ss, T value)
+    {
+        char buffer[32];
+        int32_t i = 0;
+
+        size_t num;
+        if (value < 0)
+        {
+            // TODO handle edge case: std::numeric_limits<int64_t>::min();
+            num = -value;
```

Comment:
Did you intend to negate what is already negative? Because then I think this code would be a lot clearer with:
```c++
if (value < 0)
{
    ss << '-';
}
size_t num = std::abs(value);

Reply:
This would incur two checks, but I will need to do some clever stuff anyway to handle the edge case of negative long max value which can't be represented as positive.